xilem/examples/stopwatch.rs

```diff
@@ -0,0 +1,263 @@
+                    self.added_duration
+                }
+            },
+            _ => self.added_duration,
+        }
+    }
+
+    fn update_display(&mut self) {
+        self.displayed_duration = self.get_current_duration();
+    }
+}
+
+fn get_formatted_duration(dur: &Duration) -> String {
+    let seconds = dur.as_secs_f64() % 60.0;
+    let minutes = (dur.as_secs() / 60) % 60;
+    let hours = (dur.as_secs() / 60) / 60;
+    format!("{hours}:{minutes:0>2}:{seconds:0>4.1}")
+}
+
+fn app_logic(data: &mut Stopwatch) -> impl WidgetView<Stopwatch> {
+    fork(
+        flex((
+            FlexSpacer::Fixed(5.0),
+            label(get_formatted_duration(&data.displayed_duration)).text_size(70.0),
+            flex((lap_reset_button(data), start_stop_button(data))).direction(Axis::Horizontal),
+            FlexSpacer::Fixed(1.0),
+            laps_section(data),
+            label(data.displayed_error.as_ref()),
+        ))
+        .main_axis_alignment(MainAxisAlignment::Start)
+        .cross_axis_alignment(CrossAxisAlignment::Center),
+        async_repeat(
+            |proxy| async move {
+                let mut interval = time::interval(Duration::from_millis(50));
+                loop {
+                    interval.tick().await;
+                    let Ok(()) = proxy.message(()) else {
+                        break;
+                    };
+                }
+            },
+            |data: &mut Stopwatch, ()| {
+                if data.active {
+                    data.update_display();
+                }
+            },
+        ),
+    )
+}
+
+/// Creates a list of items that shows the lap number, split time, and total cumulative time.
+fn laps_section(data: &mut Stopwatch) -> Vec<AnyFlexChild<Stopwatch>> {
+    let mut items: Vec<AnyFlexChild<Stopwatch>> = Vec::new();
+    let mut total_dur = Duration::ZERO;
+    let current_lap = data.completed_lap_splits.len();
+
+    for (i, split_dur) in data.completed_lap_splits.iter().enumerate() {
+        total_dur = total_dur.add(*split_dur);
+        items.push(single_lap(i, split_dur, &total_dur).into_any_flex());
+    }
+    let current_split_duration = data.get_current_duration().sub(total_dur);
+    let mut reversed = Vec::new();
+    reversed.push(
+        single_lap(
+            current_lap,
+            &current_split_duration,
+            &data.get_current_duration(),
+        )
+        .into_any_flex(),
+    );
+    for item in items.into_iter().rev() {
+        reversed.push(item);
+    }
+    reversed
+}
+
+fn single_lap(
+    lap_id: usize,
+    split_dur: &Duration,
+    total_dur: &Duration,
+) -> impl WidgetView<Stopwatch> {
+    flex((
+        FlexSpacer::Flex(1.0),
+        label(format!("Lap {}", lap_id + 1)).flex(1.0),
+        label(get_formatted_duration(split_dur)).flex(1.0),
+        label(get_formatted_duration(total_dur)).flex(1.0),
+    ))
+    .direction(Axis::Horizontal)
```

Comment:
I think you might be able to use `must_fill_major_axis(true)` here?

Reply:
That fixed it. Thanks!